src/components/Checkout/Checkout.js

```diff
@@ -0,0 +1,936 @@
+  const [updateProfileAlert, setUpdateProfileAlert] = useState(false);
+  const [quitEditingAlert, setQuitEditingAlert] = useState(false);
+  const [profileNameDialog, setProfileNameDialog] = useState(false);
+  // Ref used to ensure profile name loaded before submitting to airtable
+  const updatingName = useRef(false);
+
+  // empty profile for new profile creation/ resetting fields
+  const emptyProfile = {
+    'profile name': '',
+    'first name': '',
+    'last name': '',
+    phone: '',
+    email: '',
+    'address line 1': '',
+    'address line 2': '',
+    city: '',
+    state: '',
+    country: '',
+    zipcode: '',
+    'prefer delivery': false,
+    'can ACH': false,
+    'can paypal': false,
+    'can card': false,
+    'can check': false,
+  };
+
+  // current profile being used
+  const [profile, setProfile] = useState(emptyProfile);
+
+  // initial valid settings
+  const emptyValid = {
+    'first name': false,
+    'last name': false,
+    phone: false,
+    email: false,
+    'address line 1': false,
+    'address line 2': true, // always ok, but look into possible validation
+    city: false,
+    state: false,
+    country: false,
+    zipcode: false,
+  };
+
+  // valid initializer for loading past profiles (guaranteed to be valid)
+  const allValid = {
+    'first name': true,
+    'last name': true,
+    phone: true,
+    email: true,
+    'address line 1': true,
+    'address line 2': true,
+    city: true,
+    state: true,
+    country: true,
+    zipcode: true,
+  };
+
+  // validation state of each text field
+  const [valid, setValid] = useState(emptyValid);
+  const paymentValid = [profile['can ACH'], profile['can paypal'], profile['can card'], profile['can check']].filter((v) => v).length > 0;
+
+  // Fetches all necessary airtable data on start
+  // TODO: only fetch records related to user
+  useEffect(() => {
+    // fetch all the items in cart and set the subtotal
+    setSubtotal(0);
+    base('Reserved Listings').select({ view: 'Grid view' }).all().then((records) => {
+      records.map((element) => base('Listings').find(element.fields['listing id'][0], (err, record) => {
+        const currCartItemPrice = element.fields.pallets * record.fields['standard price per unit'];
+        setSubtotal(subtotal + currCartItemPrice);
+      }));
+      setItems(records);
+    });
+    // fetch all the checkout profiles
+    base('Checkout Profiles').select({ view: 'Grid view' }).all().then((records) => {
+      setPastProfiles(records);
+    });
+  }, []);
+
+  // links to success page
+  // runs on successful submission of quote request
+  const toSuccess = () => history.push('/cart/success');
+
+  // method to handle text field or checkbox input change
+  const handleChange = (event) => {
+    const { target } = event;
+    const { name } = target;
+    let value = target.type === 'checkbox' ? target.checked : target.value;
+    if (name === 'prefer delivery') { value = !profile['prefer delivery']; }
```

Comment:
Yeah! I have it set up so that if 'prefer delivery' is checked then the 'prefer pickup' checkbox is unchecked. So rather than a checkbox, I want it to act more like a toggle between delivery and pickup, but with the ui of the checkbox.